manifests/slave.pp

```diff
@@ -160,7 +160,7 @@
 
   if $labels {
     if $labels =~ Array {
-      $_combined_labels = hiera_array('jenkins::slave::labels', $labels)
+      $_combined_labels = lookup( { 'name' => 'jenkins::slave::labels', 'default_value' => $labels } )
```

Comment:
I wonder if this is really needed. The hiera key `jenkins::slave::labels` means `$labels` already contains the correct content, right? I think that you can simply drop `hiera_array` everywhere.

Reply:
Not quite.  Specifically, `hiera_array` would have been merging arrays across the hierarchy.

Reply:
You need to either set the merge behaviour as a parameter to `lookup` here, or set lookup_options data in module hiera.  See https://github.com/voxpupuli/puppet-telegraf/pull/134 for example.

Reply:
You mean something like this `lookup('jenkins::slave::labels', Any, unique, $labels)` ?

Reply:
yeah, something like is needed to preserve behaviour.  Bonus points if you can rid of it and use `lookup_options` in module hiera though.  That is more flexible as users can override the merge behaviour in their own environment hiera if they fancy.

But don't worry if you can't get that working.  Switching to `lookup()` would still be an improvement.